From cc5f0254ceb4fe8eceb289484ae298326c4cf570 Mon Sep 17 00:00:00 2001
From: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Date: Thu, 10 Feb 2022 08:59:54 +0100
Subject: [PATCH 006/841] feat(stm32mp1): default first metadata for FWU

Provide the STM32MP default metadata template.

The following command will generate the first metadata.bin file
to save in flash:
$ tools/fwu_gen_metadata/fwumd_tool.py jsonparse \
  plat/st/stm32mp1/default_metadata.json -b metadata.bin

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git dbbb5b0299f856a3d65395ab71af3d1956bee3d4 feat(stm32mp1): default first metadata for FWU]
Signed-off-by: Nicolas Toromanoff <nicolas.toromanoff@foss.st.com>
Change-Id: I43ac35f2501d13eb2f830eeee24986e2426b557e
---
 plat/st/stm32mp1/default_metadata.json | 37 ++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 plat/st/stm32mp1/default_metadata.json

diff --git a/plat/st/stm32mp1/default_metadata.json b/plat/st/stm32mp1/default_metadata.json
new file mode 100644
index 000000000..bd40f04d2
--- /dev/null
+++ b/plat/st/stm32mp1/default_metadata.json
@@ -0,0 +1,37 @@
+{
+    "copyright": "Copyright (c) 2022, STMicroelectronics - All Rights Reserved",
+    "license": "SPDX-License-Identifier: BSD-3-Clause",
+
+    "metadata": {
+        "version": 0,
+        "active_index": 0,
+        "previous_active_index": 1,
+        "img_entry": {
+            "img_0": {
+                "location": "device0",
+                "img_bank_info": {
+                    "img_0_bank_0": {
+                        "accepted": true
+                    },
+                    "img_0_bank_1": {
+                        "accepted": false
+                    }
+                }
+            }
+        }
+    },
+    "uuids": {
+        "locations": {
+            "device0": "00000000-0000-0000-0000-000000000000"
+        },
+        "entries": {
+            "img_0_bank_0": "4fd84c93-54ef-463f-a7ef-ae25ff887087",
+            "img_0_bank_1": "09c54952-d5bf-45af-acee-335303766fb3",
+            "img_0":        "19d5df83-11b0-457b-be2c-7559c13142a5"
+        }
+    },
+    "configs": {
+        "nb_fw_img": 1,
+        "nb_fw_banks": 2
+    }
+}
-- 
2.47.2

